### PR TITLE
test(): migrate tests to playwright/jest

### PR DESCRIPTION
**Pull request by ShaMan123**
## Motivation

#9329 

## Description

Working on #9329 unconvered many ubgs.
Trying to fix them made tests fail.
These tests are not human friendly, they dump endless output onto the console so they are useless.
To be percise `draggable_text` was written by me while grinding my teeth. It was hard and I knew it was incorrect to write it like that.
So I have moved  `draggable_text` to e2e as it should be, removing redundancies.
And I have moved all the `findTarget` related tests to jest.

### CODE REVIEW

File: e2e/tests/text/drag&drop/index.spec.ts-snapshots/4-events.json [hunk 1]
```diff
@@ -0,0 +1,191 @@
...
+        "isTrusted": true
+      },
+      "target": "b",
+      "subTargets": [],
+      "dragSource": "a"
+    },
+    "canvas"
+  ],
+  [
+    "dragleave",
```
Comment (ShaMan123) [inline e2e/tests/text/drag&drop/index.spec.ts-snapshots/4-events.json:35]: weird, a bug.
after dragstart nothing has entered yet

File: e2e/tests/text/drag&drop/index.spec.ts-snapshots/5-events.json [hunk 2]
```diff
@@ -0,0 +1,191 @@
...
+        "x": 120,
+        "y": 55
+      },
+      "absolutePointer": {
+        "x": 120,
+        "y": 55
+      }
+    },
+    "b"
+  ],
```
Comment (ShaMan123) [inline e2e/tests/text/drag&drop/index.spec.ts-snapshots/5-events.json:55]: here canvas says b is entering while b is saying it is leaving

**Review by ShaMan123 — COMMENTED** [review]
reverted, now only tests are here and #9329

**Review by ShaMan123 — COMMENTED** [review]
@asturur please merge this
The tests in this PR are much more freindly and reliable - I have PRs with tests that are failing, tests that are migrated here.

**Review by ShaMan123 — COMMENTED** [review]
It seems that tests are failing becuase of font differences.
We need to register the font - the code you removed from the other CI PR

File: e2e/tests/text/drag&drop/index.ts [hunk 3]
```diff
@@ -77,41 +77,100 @@ const styles = {
...
+  caller: fabric.Textbox | fabric.Canvas
+) =>
+  JSON.parse(
+    JSON.stringify([
+      type,
+      {
+        ...ev,
+        ...(pointer ? { pointer: roundPoint(pointer) } : {}),
+        ...(absolutePointer
+          ? { absolutePointer: roundPoint(absolutePointer) }
```
Comment (ShaMan123) [inline e2e/tests/text/drag&drop/index.ts:97]: scenePoint/viewportPoint

**Review by asturur — APPROVED** [review]

**Comment by asturur** [comment]
I ll make an issue with a followup for this PR since the the position of the mouse should be refactored with the proper text utils where possible.
Also one test for me doesn't pass locally

**Comment by ShaMan123** [comment]
> I ll make an issue with a followup for this PR since the the position of the mouse should be refactored with the proper text utils where possible. Also one test for me doesn't pass locally

My point of view on this is that we should use hardcoded position because it makes the test strict.
If we use a function that depends on what we are testing then the test becomes weaker, don't you think?

**Pull request merged**